usaspending_api/awards/v2/filters/matview_filters.py

```diff
@@ -249,9 +249,9 @@ def keyword_parse(keyword):
                 queryset = queryset.filter(cfda_number__in=in_query)
 
         elif key == "naics_codes":
-            in_query = [v for v in value]
-            if len(in_query) != 0:
-                queryset = queryset.filter(naics_code__in=in_query)
+            if value:
+                regex = f"^({'|'.join([str(elem) for elem in value])}).*"
```

Comment:
Doing some tests with Brian against DEV with larger datasets, this regexp query with Postgres seemed to lag a whole bunch. 
We need a broader discussion about how best to implement these types of postgres searches/matches of hierarchical selections with e.g. @kbard and @tony-sappe and @emily-brents before this code can move forward. 

We may never get it where we want it with postgres, hence providing an Elasticsearch backend impl, but we should at least be consistent with using the best-possible postgres impl.